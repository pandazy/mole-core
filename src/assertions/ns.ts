import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
