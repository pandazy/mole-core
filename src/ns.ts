export function ns(context: string): string {
  return `Mole:${context}`;
}
