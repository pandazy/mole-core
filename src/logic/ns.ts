import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
