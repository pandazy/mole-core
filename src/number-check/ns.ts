import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
