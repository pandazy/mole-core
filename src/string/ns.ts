import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
