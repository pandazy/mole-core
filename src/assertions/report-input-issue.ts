import { replaceAllBy } from '../string/replace-all';
import { printInput } from '../print-input';
import { isA, Nil } from '../type-check/is-a';

export const reportInputIssue = (message: string) => (input: any): string => {
	const valString = printInput(input);
	const instanceName = isA(Nil)(input)
		? `Nil:(${input})`
		: Object.getPrototypeOf(input).constructor.name;
	return replaceAllBy({
		'%arg%': valString,
		'%argType%': instanceName,
	})(message);
};
