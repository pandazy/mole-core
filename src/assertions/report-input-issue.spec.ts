import { reportInputIssue } from './report-input-issue';
import { ns } from './ns';

describe(ns('reportInputIssue'), () => {
	it('should return correct issue report', () => {
		expect.assertions(3);

		class CustomClass {}

		const template = '%arg% | %argType%';

		expect(reportInputIssue(template)(undefined)).toMatchSnapshot();
		expect(reportInputIssue(template)(2)).toMatchSnapshot();
		expect(reportInputIssue(template)(new CustomClass())).toMatchSnapshot();
	});
});
