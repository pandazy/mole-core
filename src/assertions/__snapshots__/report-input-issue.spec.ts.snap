// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:assertions:reportInputIssue should return correct issue report 1`] = `"undefined | Nil:(undefined)"`;

exports[`Mole:assertions:reportInputIssue should return correct issue report 2`] = `"2 | Number"`;

exports[`Mole:assertions:reportInputIssue should return correct issue report 3`] = `"{} | CustomClass"`;
